packages/cli/src/commands/devices/commands.ts

```diff
@@ -1,49 +1,81 @@
-import { flags } from '@oclif/command'
+import { Device, Command, Status } from '@smartthings/core-sdk'
 
-import { Command, CommandList } from '@smartthings/core-sdk'
+import {APICommand, SelectingInputOutputAPICommand} from '@smartthings/cli-lib'
 
-import { APICommand } from '@smartthings/cli-lib'
+import inquirer from 'inquirer'
 
 
-export default class DevicesCommands extends APICommand {
-	static description = 'execute commands on a device'
+export function buildTableOutput(this: APICommand, data: Status): string {
+	return data.status === 'success' ? 'Command executed successfully' : `Execution status = ${data.status}`
+}
 
-	static flags = {
-		...APICommand.flags,
-		data: flags.string({
-			char: 'd',
-			description: 'JSON data for command(s)',
-		}),
-	}
+export default class DeviceCommandsCommand extends SelectingInputOutputAPICommand<Command[], Status, Device> {
+	static description = "get the current status of all of a device's component's attributes"
+
+	static flags = SelectingInputOutputAPICommand.flags
+
+	static args = [
+		{
+			name: 'id',
+			description: 'the device id',
+		},
+	]
+
+	protected buildTableOutput = buildTableOutput
 
-	static args = [{
-		name: 'id',
-		description: 'the device on which you want to execute a command',
-		required: true,
-	}]
+	primaryKeyName = 'deviceId'
+	sortKeyName = 'label'
+	listTableFieldDefinitions = ['label', 'name', 'type', 'deviceId']
 
-	private async executeAndDisplay(id: string, commands: Command[]): Promise<void> {
-		await this.client.devices.executeCommands(id, commands)
+	protected async getInputFromUser(): Promise<Command[]> {
+		const cmd = (await inquirer.prompt({
+			type: 'input',
+			name: 'deviceCommand',
+			message: 'Enter device command [component:]<capability>:<command>([<arguments]>):',
+			validate: (input: string) => {
+				// switch:on
+				// switch:on()
+				// outlet1:switch:on()
+				// switchLevel:setLevel(20)
+				return new RegExp(/^([a-zA-Z0-9]+:)?[a-zA-Z0-9]+:[a-zA-Z0-9]+(\(.*\))?$/).test(input) || 'Invalid command syntax'
+			},
+		})).deviceCommand
```

Comment:
I would rather see this asked as multiple questions. If the user has to put it all in one line, why not just have them enter it on the command line?